src/Connectors/ZeroDowntimeMySqlConnector.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Daursu\ZeroDowntimeMigration\Connectors;
+
+use  Illuminate\Database\Connectors\MySqlConnector;
+
+class ZeroDowntimeMySqlConnector extends MySqlConnector
+{
+     /**
+     * Get the PDO options based on the configuration.
+     *
+     * @param  array  $config
+     * @return array
+     */
+    public function getOptions(array $config)
+    {
+        $options = $config['options'] ?? [];
+
+        if(version_compare(PHP_VERSION, '8.1.0', '>=')){
+            //Removing the percona toolkit command options, not necessary for a PDO connection
+            $options = array_filter($options, fn($option) => !is_string($option) && substr($option, 0, 2) !== '--');
```

Comment:
can you please explain why this is required?

Reply:
Sure. The library passes all  [options from the connection config](https://github.com/Daursu/laravel-zero-downtime-migration#pt-online-schema-change) to the `PDO` constructor. 
Before PHP 8.1 it was not an issue, but as of 8.1 **a `TypeError` is thrown**, since the keys of the config options (0,1,2..) interfere with the [PDO flags values'](https://www.php.net/manual/en/pdo.constants.php#pdo.constants.param-null). On the other hand as of PHP 8.1 these flags' values are strictly typed in the context of the PDO constructor options array.